### ISSUE TITLE
`predict()` hardcodes `antialias=False` resize — silent output drift for checkpoints trained on antialiased inputs (Roboflow platform Resize, default torchvision backend), no opt-out

### ISSUE THREAD

**Issue by Mateuszry** [roboflow/rf-detr#1516]
## Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

(Related: #1203 / #1206 introduced the current behaviour; #1269 aligned the export paths with it. This report is about the cases that change did not cover.)

## Bug

Since 1.9.0 (#1206), `RFDETR.predict()` resizes with a hardcoded `antialias=False`:

https://github.com/roboflow/rf-detr/blob/1.10.1/src/rfdetr/detr.py#L2608

The rationale in #1203 was to match the Albumentations training resize (`cv2.INTER_LINEAR`, no antialias). That holds for checkpoints trained with the Albumentations backend, but not for several other common training setups. For those, 1.9.0 **moved `predict()` away from the training distribution**. The public API has no way to restore the previous behaviour:

1. **The default training backend in ≥ 1.9 antialiases.** Without the `[augment]` extra, training/validation resolve to the torchvision backend, which resizes PIL images with `antialias=True`:
   https://github.com/roboflow/rf-detr/blob/1.10.1/src/rfdetr/datasets/_torchvision.py#L406
   (`coco.py` itself warns about this: *"Albumentations (cv2 INTER_LINEAR, no antialias) to torchvision (BILINEAR + antialias=True)"*.) So with a default install, a model trained with 1.10.1 is trained on antialiased inputs and served non-antialiased inputs by `predict()`.
2. **Checkpoints trained with rfdetr ≤ 1.4** used `SquareResize` → `F.resize` on PIL images, i.e. antialiased. This was already pointed out in the #1206 discussion.
3. **Models trained on the Roboflow platform with a dataset-version "Resize" preprocessing step.** Here the images are downscaled *before* any training pipeline sees them. I checked what that step produces. Downscaled 3 original photos (2930–5712 px) to 576×576 locally with several filters and compared each result to the 576×576 image exported by Roboflow for the same photo (PSNR, higher = closer):

   | Original size | torchvision bilinear `antialias=True` | PIL BILINEAR | PIL BOX | torchvision bilinear `antialias=False` (current `predict()`) | PIL NEAREST |
   |---|---|---|---|---|---|
   | 5712×4284 | **36.22 dB** | 36.22 | 35.86 | 32.69 | 32.38 |
   | 2930×1975 | **34.04 dB** | 34.04 | 33.57 | 30.43 | 29.35 |
   | 3024×4032 | 32.97 dB | 32.97 | **33.28** | 27.60 | 27.12 |

   The platform Resize is clearly antialiased. `antialias=False` is 3.5–5.4 dB further from it, and almost as far as nearest-neighbour. The model saw exactly these images during training. At inference, `predict()` downscales the same kind of photo without antialiasing.

The effect grows with the downscale factor. For COCO-sized inputs (~640 px → 576) antialiasing barely matters. For phone photos (3000–6000 px → 576, a factor of 5–10×) it matters a lot, because non-antialiased bilinear samples only a 2×2 neighbourhood out of each ~5×7–10×10 source block.

**Observed impact** (15 fine-tuned `RFDETRMedium` checkpoints trained on the Roboflow platform / rfdetr ≤ 1.4, 8 field photos, `threshold=0.5`, same torch/torchvision), 1.4.2 → 1.10.1:

- 4 of the 21 model×image pairs with any detection returned a different set of detections: 4 detections disappeared and 1 new one appeared, all with confidence 0.50–0.65.
- Surviving detections shifted confidence by up to 0.22 (e.g. 0.86 → 0.75, 0.61 → 0.82) and moved boxes by up to 35 px.
- If force `antialias=True` (or pre-resize with antialiasing, see below), 1.10.1 reproduces the 1.4.2 outputs exactly: all 21 pairs match, max |Δconf| < 1e-4. The remaining box differences are only the edge clamping added in 1.7.1.

So the change is a silent accuracy regression for these checkpoints. The only way to opt out today is to pre-resize outside the library and rescale boxes afterwards, which depends on the internal `model.model.resolution`.

## Environment

- RF-DETR: 1.10.1 (baseline for comparison: 1.4.2; 1.8.3 matches 1.4.2 exactly)
- OS: macOS 26.5 (arm64, Apple Silicon)
- Python: 3.12.1
- PyTorch: 2.8.0, torchvision: 0.23.0
- CUDA/cuDNN: n/a
- Hardware: Apple Silicon (default device selection); the effect is device-independent, as it comes from preprocessing

## Minimal Reproducible Example

This uses the public COCO `RFDETRMedium` weights and any large photo (≥ ~3000 px on the long side):

```python
import sys

import torchvision.transforms.functional as F
from PIL import Image
from rfdetr import RFDETRMedium

model = RFDETRMedium()
resolution = model.model.resolution

image = Image.open(sys.argv[1]).convert("RGB")
width, height = image.size

default = model.predict(image, threshold=0.3)

pre_resized = F.resize(F.to_tensor(image), [resolution, resolution], antialias=True).clamp(0, 1)
antialiased = model.predict(pre_resized, threshold=0.3)
antialiased.xyxy *= [width / resolution, height / resolution] * 2

print("predict() default (antialias=False):", sorted((int(c), round(float(s), 3)) for c, s in zip(default.class_id, default.confidence)))
print("antialiased pre-resize             :", sorted((int(c), round(float(s), 3)) for c, s in zip(antialiased.class_id, antialiased.confidence)))
```

Output on a 4032 px photo (COCO weights):

```
predict() default (antialias=False): [(44, 0.316), (44, 0.34), (44, 0.345), (44, 0.347), (44, 0.379), (44, 0.556)]
antialiased pre-resize             : [(44, 0.308), (44, 0.367), (44, 0.458), (78, 0.355)]
```

## Additional

**Proposed fix:** make the resize antialias configurable and keep the current default, so nothing changes for existing users:

```python
def predict(self, images, threshold=0.5, shape=None, patch_size=None,
            include_source_image=True, antialias: bool = False, **kwargs):
    ...
    batch_tensor = torch.stack([F.resize(t, resize_to, antialias=antialias) for t in processed_images])
```

- The default `False` keeps the #1206 behaviour and ONNX/TFLite parity (#1269).
- `antialias=True` restores the ≤ 1.8 behaviour for checkpoints trained on antialiased inputs.
- The docs could state which setting matches which training setup: Albumentations backend → `False`; torchvision backend / rfdetr ≤ 1.4 / Roboflow-platform Resize → `True`.
- Optional follow-up (out of scope for a minimal fix): let the default follow the training backend, e.g. recorded in the checkpoint or in `ModelContext`, so that `predict()` matches training without user input.

## Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!

**Comment by Mateuszry** [comment]
Follow-up with results from a larger sample. The numbers in the original report came from only 8 images, and they understate the effect considerably.

**Setup:**

- 1,207 field photos from 56 categories. Most are phone photos of 2,000–6,000 px, stretched to 576×576 (512×512 for one model) by `predict()`.
- The same 15 fine-tuned `RFDETRMedium` checkpoints, trained on antialiased inputs (Roboflow-platform Resize / rfdetr ≤ 1.4).
- The same torch 2.8.0 / torchvision 0.23.0 in every run. Only the `rfdetr` version changes.
- Predictions were recorded down to `threshold=0.3`, then compared at 0.5 and 0.85.
- Detections were matched across versions by class and IoU ≥ 0.5. A detection counts as lost or gained when it clears the threshold in only one of the two versions.

### 1.8.3 vs 1.4.2 (control: last release before #1206)

| Threshold | Detections | Lost / gained | Model×image pairs with a different result | \|Δconf\| p50 / p95 / max |
|---|---|---|---|---|
| 0.5 | 5914 → 5914 | 0 / 0 | **0 / 4537** | 0.0000 / 0.0000 / 0.0053 |
| 0.85 | 3214 → 3214 | 0 / 0 | **0 / 2885** | 0.0000 / 0.0000 / 0.0009 |

The results are identical for every one of the 15 models. The only box differences come from the edge clamping added in 1.7.1: 866 of the 873 coordinates that differ by more than 0.5 px sit exactly on the image border.

### 1.10.1 vs 1.4.2

| Threshold | Detections | Lost / gained | Model×image pairs with a different result | \|Δconf\| p50 / p95 / max | \|Δbox\| p95 |
|---|---|---|---|---|---|
| 0.5 | 6283 → 6048 | 720 / 485 | **1046 / 5090 (21%)** | 0.009 / 0.146 / 0.403 | 28 px |
| 0.85 | 3422 → 3297 | 296 / 171 | **448 / 3222 (14%)** | 0.002 / 0.033 / 0.099 | 24 px |

Per model at `threshold=0.85`, sorted by impact:

| Model | Detections 1.4.2 → 1.10.1 | Pairs with a different result |
|---|---|---|
| A (small text labels) | 172 → 150 | 39 / 95 (41%) |
| B (10 classes, 512 px) | 210 → 190 | 89 / 240 (37%) |
| C (small text labels) | 59 → 55 | 19 / 56 (34%) |
| D | 195 → 186 | 38 / 152 (25%) |
| E | 316 → 295 | 48 / 257 (19%) |
| F | 289 → 266 | 46 / 248 (19%) |
| G | 767 → 753 | 106 / 788 (13%) |
| H | 141 → 133 | 18 / 146 (12%) |
| I | 80 → 78 | 10 / 84 (12%) |
| J | 263 → 258 | 19 / 268 (7%) |
| K | 19 → 20 | 1 / 20 (5%) |
| L | 86 → 87 | 3 / 88 (3%) |
| M (large objects) | 734 → 735 | 12 / 689 (2%) |
| N | 78 → 78 | 0 / 78 (0%) |
| O (large objects) | 13 → 13 | 0 / 13 (0%) |

### Takeaways

- **The drift starts exactly at 1.9.0.** 1.8.3 matches 1.4.2 bit-for-bit in practice. In the original report, forcing `antialias=True` in 1.10.1 (or pre-resizing with antialiasing) brought the results back into agreement with 1.4.2. So the drift is fully explained by the resize change in #1206 and not by any other change between 1.4 and 1.10.
- **The drift is systematic, not random.** At every threshold 1.10.1 loses more detections than it gains, −3.7% at 0.85. The models most affected are the ones looking for small objects or fine detail (text labels). Models detecting large objects are barely affected. This is what aliasing would predict: at a downscale factor of 5–10×, non-antialiased bilinear samples 4 pixels out of every ~35–100.
- **This matters at production thresholds**, not only near the decision boundary. At 0.85, one model×image pair in seven changes its output.

For now I'm pinning `rfdetr>=1.8.3,<1.9`, but I would be happy to open the PR for the `antialias` parameter proposed above (default `False`, so the current behavior is kept)
